lib/SPIRV/libSPIRV/SPIRVInstruction.h

```diff
@@ -1877,7 +1877,6 @@ class SPIRVLifetime : public SPIRVInstruction {
     assert(Obj->getStorageClass() == StorageClassFunction &&
         "Invalid storage class");
     assert(Obj->getType()->isTypePointer() &&
-        Obj->getType()->getPointerElementType()->isTypeInt() &&
         "Object type must be an integer type scalar");

```

Comment:
Please, update the message to avoid confusion.
"Objects type must be a pointer".